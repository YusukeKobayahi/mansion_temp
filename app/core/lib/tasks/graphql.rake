# frozen_string_literal: true

module GraphqlTask
  extend Rake::DSL

  namespace :graphql do
    namespace :consumer do
      task schema_dump: :environment do
        require 'graphql/rake_task'

        GraphQL::RakeTask.new(
          load_schema: lambda { |_task|
            require File.expand_path('../../app/graphql/consumer/inno_schema', __dir__)
            Consumer::InnoSchema
          },
          directory: './graphql/consumer',
        )
        Rake::Task['graphql:schema:idl'].invoke
      end
    end
  end
end
