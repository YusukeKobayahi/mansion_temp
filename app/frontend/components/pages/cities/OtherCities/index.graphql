fragment MsLibraryCitiesOtherCities on City {
  name
  prefecture {
    name
    jisCode
    cities {
      name
      jisCode
      mansionCount
    }
  }
}
