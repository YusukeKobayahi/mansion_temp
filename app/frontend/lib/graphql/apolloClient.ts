// 参考
// https://github.com/vercel/next.js/tree/canary/examples/with-apollo

import { useMemo } from "react";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
import { createHttpLink } from "apollo-link-http";
import { from } from "apollo-link";
import { createFragmentArgumentLink } from "apollo-link-fragment-argument";
import { mergeDeep, relayStylePagination } from "@apollo/client/utilities";
import { isSSR } from "~/lib/utils";

/* eslint-disable @typescript-eslint/no-explicit-any -- キャッシュの型を事前に決められないため */
type CacheState = Record<string, any> | null;
/* eslint-enable @typescript-eslint/no-explicit-any */

/* eslint-disable @typescript-eslint/naming-convention -- GraphQLのタイプ名 (このconventionに従わない)を指定したいので */
const createCache = () =>
  new InMemoryCache({
    typePolicies: {
      Mansion: {
        keyFields: ["uniqueCode"],
        fields: {
          salesHistories: relayStylePagination(),
        },
      },
    },
  });
/* eslint-enable @typescript-eslint/naming-convention */

const link = from([
  createFragmentArgumentLink(),
  createHttpLink({
    uri: `${process.env.NEXT_PUBLIC_CORE_API_URL}/consumer/graphql`,
    fetchOptions: {
      mode: "cors",
    },
  }),
]);

function createApolloClient() {
  return new ApolloClient({
    ssrMode: isSSR(),
    // See: https://github.com/apollographql/apollo-link/issues/1258#issuecomment-604031488
    // TODO: @apollo/clientバージョン上げたら直るかも
    link: (link as unknown) as ApolloLink,
    cache: createCache(),
    defaultOptions: {
      watchQuery: {
        // SSR時にuseQueryによってリクエストしてレンダリングする場合、
        // getServerSidePropsからキャッシュが渡っている必要がある。(でないと描画されない)
        // そのため、SSR時は常にcacheから取得されるべきで、実装意図を明確にするためPolicyも設定しておく
        // (エラーになるわけではない)
        fetchPolicy: isSSR() ? "cache-only" : "cache-first",
      },
    },
  });
}

let apolloClient: ReturnType<typeof createApolloClient>;

export function initializeApollo(
  initialState: CacheState = null
): ApolloClient<CacheState> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = mergeDeep(initialState, existingCache);

    // Restore the cache with the merged data
    _apolloClient.cache.restore(data);
  }
  // For SSG and SSR always create a new Apollo Client
  if (isSSR()) return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: CacheState): ApolloClient<CacheState> {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
