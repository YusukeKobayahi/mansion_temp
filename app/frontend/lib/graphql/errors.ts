export interface GraphQLError {
  graphQLErrors?: {
    message: string;
    extensions?: {
      code: string;
    };
  }[];
}

export const isNotFoundError = (err: GraphQLError): boolean => {
  const graphQLErrors = err.graphQLErrors;
  if (graphQLErrors == null) return false;

  return graphQLErrors[0]?.extensions?.code === "NOT_FOUND";
};
