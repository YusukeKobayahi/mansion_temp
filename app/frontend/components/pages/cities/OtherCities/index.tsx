import { MsLibraryCitiesOtherCitiesFragment } from "~/graphql/generated";

import styles from "~/components/pages/cities/OtherCities/index.module.scss";
import BoxHeading from "~/components/commons/headings/BoxHeading";
import LinkBlock from "~/components/commons/LinkBlock";

export interface OtherCitiesProps {
  data: MsLibraryCitiesOtherCitiesFragment;
}

const OtherCities: React.FC<OtherCitiesProps> = ({
  data,
}: OtherCitiesProps) => {
  const items = data.prefecture.cities.map((city) => (
    <LinkBlock
      key={city.jisCode}
      name={city.name}
      id={city.jisCode}
      mansionCount={city.mansionCount}
      link="cities"
      eventParams={{
        action: "msl_area_click",
        category: "msl_ichiran",
        label: "msl_syuhen",
        value: 1,
      }}
    />
  ));
  return (
    <section className={styles.container}>
      <h2>
        <BoxHeading>
          {data.prefecture.name}
          {data.name}周辺エリアから中古マンションを探す
        </BoxHeading>
      </h2>
      <ul className={styles.list}>{items}</ul>
    </section>
  );
};

export default OtherCities;
