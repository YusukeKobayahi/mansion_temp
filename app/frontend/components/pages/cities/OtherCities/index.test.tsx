import { render } from "@testing-library/react";
import mockData from "~/components/pages/cities/fixtures/mockData.json";
import { CitiesQuery } from "~/graphql/generated";
import OtherCities from "~/components/pages/cities/OtherCities";

describe("components/pages/cities/OtherCities", () => {
  it("正常に表示される", () => {
    const mock = (mockData as unknown) as CitiesQuery;
    const { asFragment } = render(<OtherCities data={mock.cities[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
