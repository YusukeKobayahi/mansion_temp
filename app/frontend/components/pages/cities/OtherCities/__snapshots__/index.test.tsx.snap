// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/cities/OtherCities 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2>
      <div
        class="heading"
      >
        神奈川県横浜市鶴見区周辺エリアから中古マンションを探す
      </div>
    </h2>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14101"
        >
          横浜市鶴見区
          <span
            class="count"
          >
            (116件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14102"
        >
          横浜市神奈川区
          <span
            class="count"
          >
            (106件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14103"
        >
          横浜市西区
          <span
            class="count"
          >
            (94件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14104"
        >
          横浜市中区
          <span
            class="count"
          >
            (121件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14105"
        >
          横浜市南区
          <span
            class="count"
          >
            (85件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14106"
        >
          横浜市保土ケ谷区
          <span
            class="count"
          >
            (29件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14107"
        >
          横浜市磯子区
          <span
            class="count"
          >
            (62件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14108"
        >
          横浜市金沢区
          <span
            class="count"
          >
            (23件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14109"
        >
          横浜市港北区
          <span
            class="count"
          >
            (94件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14110"
        >
          横浜市戸塚区
          <span
            class="count"
          >
            (70件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14111"
        >
          横浜市港南区
          <span
            class="count"
          >
            (42件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14112"
        >
          横浜市旭区
          <span
            class="count"
          >
            (25件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14113"
        >
          横浜市緑区
          <span
            class="count"
          >
            (40件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14114"
        >
          横浜市瀬谷区
          <span
            class="count"
          >
            (4件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14115"
        >
          横浜市栄区
          <span
            class="count"
          >
            (21件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14116"
        >
          横浜市泉区
          <span
            class="count"
          >
            (18件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14117"
        >
          横浜市青葉区
          <span
            class="count"
          >
            (94件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14118"
        >
          横浜市都筑区
          <span
            class="count"
          >
            (108件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14131"
        >
          川崎市川崎区
          <span
            class="count"
          >
            (120件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14132"
        >
          川崎市幸区
          <span
            class="count"
          >
            (50件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14133"
        >
          川崎市中原区
          <span
            class="count"
          >
            (100件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14134"
        >
          川崎市高津区
          <span
            class="count"
          >
            (35件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14135"
        >
          川崎市多摩区
          <span
            class="count"
          >
            (37件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14136"
        >
          川崎市宮前区
          <span
            class="count"
          >
            (100件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14137"
        >
          川崎市麻生区
          <span
            class="count"
          >
            (73件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14151"
        >
          相模原市緑区
          <span
            class="count"
          >
            (8件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14152"
        >
          相模原市中央区
          <span
            class="count"
          >
            (31件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14153"
        >
          相模原市南区
          <span
            class="count"
          >
            (56件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14201"
        >
          横須賀市
          <span
            class="count"
          >
            (40件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14203"
        >
          平塚市
          <span
            class="count"
          >
            (36件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14204"
        >
          鎌倉市
          <span
            class="count"
          >
            (42件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14205"
        >
          藤沢市
          <span
            class="count"
          >
            (94件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14206"
        >
          小田原市
          <span
            class="count"
          >
            (21件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14207"
        >
          茅ヶ崎市
          <span
            class="count"
          >
            (45件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14208"
        >
          逗子市
          <span
            class="count"
          >
            (4件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14210"
        >
          三浦市
          <span
            class="count"
          >
            (1件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14211"
        >
          秦野市
          <span
            class="count"
          >
            (4件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14212"
        >
          厚木市
          <span
            class="count"
          >
            (26件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14213"
        >
          大和市
          <span
            class="count"
          >
            (52件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14214"
        >
          伊勢原市
          <span
            class="count"
          >
            (6件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14215"
        >
          海老名市
          <span
            class="count"
          >
            (23件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14216"
        >
          座間市
          <span
            class="count"
          >
            (16件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14301"
        >
          三浦郡葉山町
          <span
            class="count"
          >
            (4件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14341"
        >
          中郡大磯町
          <span
            class="count"
          >
            (3件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14342"
        >
          中郡二宮町
          <span
            class="count"
          >
            (2件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14366"
        >
          足柄上郡開成町
          <span
            class="count"
          >
            (2件)
          </span>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="link"
          href="/cities/14382"
        >
          足柄下郡箱根町
          <span
            class="count"
          >
            (8件)
          </span>
        </button>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
